File: 3000-maximum-area-of-longest-diagonal-rectangle/3000-maximum-area-of-longest-diagonal-rectangle.js
var areaOfMaxDiagonal = function (dimensions) {
    let dimension = [];
    let diagonal = 0;
    for (d of dimensions) {
        let temp = d[0] * d[0] + d[1] * d[1];
        if (temp > diagonal) {
            diagonal = temp;
            dimension = d;
        } else if (temp === diagonal) {
            if ((d[0] * d[1]) > (dimension[0] * dimension[1])) {
                dimension = d;
            }
        }
    }
    return dimension[0] * dimension[1];
};

File: 3000-maximum-area-of-longest-diagonal-rectangle/3000-maximum-area-of-longest-diagonal-rectangle.cs
public class Solution {
    public int AreaOfMaxDiagonal(int[][] dimensions) {
        int max_area = 0;
        int max_diago = 0;
        for(int i = 0; i < dimensions.Length; i++){
            int t = dimensions[i][0]*dimensions[i][0]+dimensions[i][1]*dimensions[i][1];
            if(t > max_diago || (t == max_diago && max_area < dimensions[i][0]*dimensions[i][1])){
                max_diago = t;
                max_area = dimensions[i][0]*dimensions[i][1];
            }
        }
        return max_area;
    }
}